gen/calc.py

```diff
@@ -15,6 +15,21 @@
 from pkgpanda.build import hash_checkout
 
 
+def validate_one_of(val, valid_values) -> None:
+    """Test if object `val` is a member of container `valid_values`.
+    Raise a ValueError if it is not a member. The exception message contains
+    both, the representation (__repr__) of `val` as well as the representation
+    of all items in `valid_values`.
+    """
+    if val not in valid_values:
+        options_string = ', '.join("'{}'".format(v) for v in valid_values)
+        raise AssertionError("Must be one of {}. Got '{}'.".format(options_string, val))
```

Comment:
The gen library is particular about this being an AssertionError, like how all the other things in here use "assert" to check things.

https://github.com/dcos/dcos/blob/b94e399e211da753a582c5d0a5748e73b96fecf7/gen/__init__.py#L337